### PR TITLE
functions to estimate low-hanging fruit indicators. Those that use standard output as input.

### PR DESCRIPTION
**Pull request by randrescastaneda**
Hi @tonyfujs , 

In this PR I am adding the necessary function to estimated low-hanging indicators using as input the standard PIP output. This PR works in conjunction with PR [343](https://github.com/PIP-Technical-Team/pipapi/pull/343) in the pipapi package.  All the functions have been created by @zander-prinsloo . I am only integrating them into wbpip and pipapi

Best,

### CODE REVIEW

File: R/low-hanging_indicators.R [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+#' @examples
+#' if (suppressWarnings(require(pipr))) {
+#'   pip <- get_stats()
+#'   get_lh_number_poor(
+#'     pip$headcount,
+#'     pip$pop) |>
+#'     head()
+#' }
+get_lh_number_poor <- function(headcount, pop){
+
```
Comment (tonyfujs) [inline R/low-hanging_indicators.R]: I think we can remove the the `_lh` part in all these functions' names.

File: R/low-hanging_indicators.R [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+  # Input checks
+  stopifnot(is.numeric(headcount) & is.vector(headcount)) # numeric vector
+  stopifnot(is.numeric(pop) & is.vector(pop)) # numeric vector
+  stopifnot(length(headcount) == length(pop)) # vectors of same length
+  if (any(headcount < 0) | any(pop < 0))
+    warning("both headcount and pop should be positive")
+
```
Comment (tonyfujs) [inline R/low-hanging_indicators.R]: Also valid for all the new functions introduced here: I don't think these checks are required here, as not of these functions are supposed to be used directly by the end user. They should be either:
- Moved to the ingestion pipeline when relevant
- Moved one level above in the functions supposed to be used interactively (new high-level package)

This is mostly to avoid these checks to happen live, for performance consideration.

File: tests/testthat/test-low-hanging_indicators.R [hunk 3]
```diff
@@ -0,0 +1,272 @@
...
+  expect_equal(output, hc*pop)
+
+  # expectation 2
+  hc <- rep(c(0.1, 0.2), 5)
+  pop <- sample(c(1:100), size = 10, replace = T)
+  output <- get_lh_number_poor(headcount = hc, pop = pop)
+  expect_equal(output, hc*pop)
+
+
+})
```
Comment (tonyfujs) [inline tests/testthat/test-low-hanging_indicators.R]: These tests should test edge cases. For instance hc <- -0.2

What should be this function expected behavior in case of negative insputs?

**Review by tonyfujs — COMMENTED** [review]
Thanks @randrescastaneda and @zander-prinsloo 

This looks great. I've made a few comments. Please take a look and let me know what you think, especially on the "input checks" part as we need have a general rule on how to arbitrate these trade-offs.

File: R/low-hanging_indicators.R [hunk 4]
```diff
@@ -0,0 +1,262 @@
...
+  # Input checks
+  stopifnot(is.numeric(headcount) & is.vector(headcount)) # numeric vector
+  stopifnot(is.numeric(povgap) & is.vector(povgap)) # numeric vector
+  stopifnot(is.numeric(povline) & is.vector(povline)) # numeric vector
+  stopifnot(length(headcount) == length(povline)) # vectors of same length
+  stopifnot(length(headcount) == length(povgap)) # vectors of same length
+
+  # Average Shortfall
+  av_sf <- povline*povgap/headcount # z times P1/P0
```
Comment (zander-prinsloo) [inline R/low-hanging_indicators.R]: Apologies, this is not the correct version of the function @randrescastaneda. In [this](https://github.com/PIP-Technical-Team/wbpip/commit/bead5388986751da72c93a6fc264ab23fc3d4910) commit in the `low-hanging_fruit_indicators` branch of wbpip I updated the function.

File: R/low-hanging_indicators.R (hunk 1 above)
Comment (randrescastaneda) [reply R/low-hanging_indicators.R]: Done

File: R/low-hanging_indicators.R (hunk 2 above)
Comment (randrescastaneda) [reply R/low-hanging_indicators.R]: Hi @tonyfujs, 

Sure. Those checks can be removed, but there are some other checks that maybe we should add to these functions. Say for instance the numb of poor. For any obs where either the headcount or the population is negative, the number of poor should be `NA`. For this case in particular, should we add checks for negative numbers? If we should not, we would be assuming that the output from PIP is correct. But, if we should, then we could also add several other checks like making sure that each vector is within its corresponding range. For example, we want headcount to be between 0 and 1.

What do you think? 


CC: @zander-prinsloo

File: tests/testthat/test-low-hanging_indicators.R (hunk 3 above)
Comment (randrescastaneda) [reply tests/testthat/test-low-hanging_indicators.R]: Hi @tonyfujs , 
Agree. We will remove them, but then it is necessary to create the defenses (or check) mentioned in the comment [above](https://github.com/PIP-Technical-Team/wbpip/pull/225#discussion_r1254820207). Otherwise, we won't be able to check for those edge cases. What do you think?

cc: @zander-prinsloo

File: R/low-hanging_indicators.R (hunk 4 above)
Comment (randrescastaneda) [reply R/low-hanging_indicators.R]: Thanks, @zander-prinsloo.
I already incorporated your changes.

File: R/low-hanging_indicators.R (hunk 2 above)
Comment (randrescastaneda) [reply R/low-hanging_indicators.R]: Hi @tonyfujs , 
We did as agreed on Team chat.

File: tests/testthat/test-low-hanging_indicators.R (hunk 3 above)
Comment (randrescastaneda) [reply tests/testthat/test-low-hanging_indicators.R]: Done

**Comment by randrescastaneda** [comment]
Hi @tonyfujs , 

All your comments have been addressed, mainly by @zander-prinsloo  (thank you!). I also fix a few things in the package to make sure `devtools::check()`s pass.

image
